src/http.ts

```diff
@@ -78,32 +82,39 @@ class Http {
       const xhr = new XMLHttpRequest();
       xhr.open(method, url, true);
 
-      // set standard headers
-      xhr.setRequestHeader('Authorization', publishableKey);
-      xhr.setRequestHeader('Content-Type', 'application/json');
-      xhr.setRequestHeader('X-Radar-Device-Type', 'Web');
-      xhr.setRequestHeader('X-Radar-SDK-Version', SDK_VERSION);
-
-      // set passed custom headers if present
-      if (headers) {
-        Object.keys(headers).forEach(key => {
-          const val = headers[key];
-          xhr.setRequestHeader(key, val);
-        });
-      }
+      const defaultHeaders = {
+        'Authorization': publishableKey,
+        'Content-Type': 'application/json',
+        'X-Radar-Device-Type': 'Web',
+        'X-Radar-SDK-Version': SDK_VERSION,
+      };
 
-      // set config custom headers if present
+      // set custom config headers if present
+      let configHeaders: { [key: string]: string } = {};
       if (typeof options.getRequestHeaders === 'function') {
-        const headers: { [key: string]: string } = options.getRequestHeaders();
-        Object.keys(headers || {}).forEach((key) => {
-          xhr.setRequestHeader(key, headers[key]);
-        });
+        configHeaders = options.getRequestHeaders();
+      }
+
+      // combines default headers with custom headers and config headers
+      const allHeaders = Object.assign(defaultHeaders, configHeaders, headers);
+
+      // set headers
+      Object.keys(allHeaders).forEach((key) => {
+        xhr.setRequestHeader(key, allHeaders[key]);
+      });
+
+      if (responseType) {
+        xhr.responseType = responseType;
       }
 
       xhr.onload = () => {
         let response: any;
         try {
-          response = JSON.parse(xhr.response);
+          if (xhr.responseType === 'blob') {
+            response = { code: xhr.status, data: xhr.response };
+          } else {
+            response = JSON.parse(xhr.response);
+          }
```

Comment:
so ideally this is:
```suggestion
          if (xhr.responseType === 'json') {
            response = JSON.parse(xhr.response);
          } else {
            response = { code: xhr.status, data: xhr.response };
          }
```

but when i wrote this initially i wanted to minimize logic changes so we don't introduce new sweeping bugs